### PR TITLE
Use windows-sys and fix feature flag

### PR DESCRIPTION
**Pull request by clemenswasser**
Implements #639
Also fixes the `windows_process_extensions_main_thread_handle` feature toggle so that It now actually works 😄

### CODE REVIEW

File: src/timer/mod.rs [hunk 1]
```diff
@@ -83,7 +86,7 @@ pub fn execute_and_measure(mut command: Command) -> Result<TimerResult> {
         use std::os::windows::process::CommandExt;
 
         // Create a suspended process
-        command.creation_flags(4);
+        command.creation_flags(CREATE_SUSPENDED);
```
Comment (sharkdp) [inline src/timer/mod.rs:89]: :+1: 

Maybe we can remove the comment now? (Or explain what a suspended process is?)

File: src/timer/windows_timer.rs [hunk 2]
```diff
@@ -48,24 +52,24 @@ pub struct CPUTimer {
...
         let job_object = unsafe { CreateJobObjectW(ptr::null_mut(), ptr::null_mut()) };
-        assert!(!job_object.is_null(), "CreateJobObjectW failed");
+        assert!(job_object != 0, "CreateJobObjectW failed");
 
         // SAFETY: The job object handle is valid
         let ret = unsafe { AssignProcessToJobObject(job_object, child_handle) };
         assert!(ret != 0, "AssignProcessToJobObject failed");
 
-        #[cfg(windows_process_extensions_main_thread_handle)]
+        #[cfg(feature = "windows_process_extensions_main_thread_handle")]
```
Comment (sharkdp) [inline src/timer/windows_timer.rs:65]: If I understand/recall correctly, the intention was that this version could be used for nightly-compiler builds (with `windows_process_extensions_main_thread_handle` additionally activated), and once that feature is stabilized, we can remove the alternative `not(feature = …)` version below and make this the main code path.

**Review by sharkdp — APPROVED** [review]
Very nice. A clear improvement.

File: src/timer/mod.rs (hunk 1 above)
Comment (clemenswasser) [reply src/timer/mod.rs:89]: I hope the explanation helps.

File: src/timer/windows_timer.rs (hunk 2 above)
Comment (clemenswasser) [reply src/timer/windows_timer.rs:65]: Yes, this is the plan 👍

**Pull request merged**